File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
  const typeCounts = {}
  
  clothes.forEach(item => {
    const type = item[1]
    typeCounts[type] = (typeCounts[type] || 0) + 1
  });
  
  let totalCombinations = 1
    
  for (let type in typeCounts) {
    totalCombinations *= (typeCounts[type] + 1)
  }
  
  return totalCombinations - 1
}

File: 프로그래머스/2/42578. 의상/의상.py
import collections

def solution(clothes):
    type_counts = collections.defaultdict(int)
    
    for item in clothes:
        type_ = item[1]
        type_counts[type_] += 1
        
    total_combinations = 1
    
    for count in type_counts.values():
        total_combinations *= (count + 1)
        
    return total_combinations - 1